src/components/Personalization/dom-actions/dom/selectNodesWithShadow.js

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2021 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import querySelectorAll from "./querySelectorAll";
+import { cleanupPrefix, splitWithShadow } from "./helperForShadow";
+
+export default (context, selector) => {
+  const parts = splitWithShadow(selector);
+
+  if (parts.length < 2) {
+    return querySelectorAll(context, selector);
+  }
+
+  // split the selector into parts separated by :shadow pseudo-selectors
+  // find each subselector element based on the previously selected node's shadowRoot
+  let parent = context;
+  for (let i = 0; i < parts.length; i += 1) {
+    const part = cleanupPrefix(parts[i]);
```

Comment:
I suspect I may be misunderstanding something here, but couldn't this result in invalid matches? For example, given the selector `BUY-NOW-BUTTON:nth-of-type(2):shadow > .logo`, it seems stripping the `>` would result in us looking for `.logo` at any descendent level within the buy now button rather than a direct descendent.

Reply:
The polyfill I mentioned above won't work, as there's no way to set an attribute on a `shadowRoot`. Anyhow, turns out custom elements and Shadow DOM aren't supported in IE in any case, so there's no need to also add support for `:scope`, since pages with custom elements and Shadow DOM will be broken in IE anyway.